src/wallet/wallet2.cpp

```diff
@@ -4236,6 +4410,329 @@ static uint32_t get_count_above(const std::vector<wallet2::transfer_details> &tr
+    // {
+    //   td.m_mask = rct::identity();
+    //   td.m_rct = true;
+    // }
+    else
+    {
+      td.m_mask = rct::identity();
+      td.m_rct = false;
+    }
+    if(!spent)
+      set_unspent(m_transfers.size()-1);
+    m_key_images[td.m_key_image] = m_transfers.size()-1;
+    m_pub_keys[td.get_public_key()] = m_transfers.size()-1;
+    if(spent)
+      MTRACE("Spent output "<< o.amount << " xmr added to m_transfers tx_hash: " << o.tx_hash);
+    else
+      MTRACE("Unspent output "<< o.amount << " xmr added to m_transfers tx_hash: " << o.tx_hash);      
+  }
+  MTRACE("dumping m_transfers");
+  for(auto &t: m_transfers)
+  {
+    if(t.m_spent)
+      MTRACE("Spent " << t.amount() << " xmr public_key: " << string_tools::pod_to_hex(t.m_txid));
+    else
+      MTRACE("Unspent " << t.amount() << " xmr public_key: " << string_tools::pod_to_hex(t.m_txid));    
+  }
+
+}
+
+bool wallet2::light_wallet_get_address_info(cryptonote::COMMAND_RPC_GET_ADDRESS_INFO::response &response)
+{
+  MTRACE(__FUNCTION__);
+  
+  cryptonote::COMMAND_RPC_GET_ADDRESS_INFO::request request;
+  
+  request.address = get_account().get_public_address_str(m_testnet);
+  request.view_key = string_tools::pod_to_hex(get_account().get_keys().m_view_secret_key);
+  m_daemon_rpc_mutex.lock();
+  bool r = epee::net_utils::invoke_http_json("/get_address_info", request, response, m_http_client, std::chrono::seconds(30), "POST");
+  m_daemon_rpc_mutex.unlock();
+  THROW_WALLET_EXCEPTION_IF(!r, error::no_connection_to_daemon, "get_address_info");
+  return true;
+}
+
+void wallet2::light_wallet_get_address_txs()
+{
+  MDEBUG("Refreshing light wallet");
+  
+  cryptonote::COMMAND_RPC_GET_ADDRESS_TXS::request ireq;
+  cryptonote::COMMAND_RPC_GET_ADDRESS_TXS::response ires;
+  
+  ireq.address = get_account().get_public_address_str(m_testnet);
+  ireq.view_key = string_tools::pod_to_hex(get_account().get_keys().m_view_secret_key);
+  m_daemon_rpc_mutex.lock();
+  bool r = epee::net_utils::invoke_http_json("/get_address_txs", ireq, ires, m_http_client, std::chrono::seconds(30), "POST");
+  m_daemon_rpc_mutex.unlock();
+  THROW_WALLET_EXCEPTION_IF(!r, error::no_connection_to_daemon, "get_address_txs");
+  
+  // Store blockchain info
+  m_light_wallet_scanned_block_height = ires.scanned_block_height;
+  m_light_wallet_blockchain_height = ires.blockchain_height;
+  
+  // Create searchable vectors
+  std::vector<crypto::hash> payments_txs;
+  for(const auto &p: m_payments)
+    payments_txs.push_back(p.second.m_tx_hash);
+  std::vector<crypto::hash> unconfirmed_payments_txs;
+  for(const auto &up: m_unconfirmed_payments)
+    unconfirmed_payments_txs.push_back(up.second.m_tx_hash);
+
+
+  uint64_t total_received, total_sent;
+  for (const auto &t: ires.transactions) {
+    total_received = (t.total_received.empty()) ? 0 :  boost::lexical_cast<uint64_t>(t.total_received);
+    total_sent = (t.total_sent.empty()) ? 0 :  boost::lexical_cast<uint64_t>(t.total_sent);
```

Comment:
Don't know. but these casts won't be needed when backend switch to numerical values.